packages/patternfly-4/react-core/src/components/Modal/Modal.js

```diff
@@ -23,7 +24,9 @@ const propTypes = {
   /** A callback for when the close button is clicked */
   onClose: PropTypes.func,
   /** Creates a large version of the Modal */
-  isLarge: PropTypes.bool
+  isLarge: PropTypes.bool,
+  /** React application root element */
+  reactRoot: PropTypes.instanceOf(safeHTMLElement).isRequired
```

Comment:
could we default to document.body and make it optional?

Reply:
I had tested several implementations before to see what worked across the different screen reader / browser combinations we test in. The attributes `role="dialog"` and `aria-modal` don't seem to be supported as described by wai-aria. When I test their solution using JAWS / Chrome, focus is not trapped. It also is not trapped using [focus-trap-react](https://github.com/davidtheclark/focus-trap-react). 

Solutions that did work were:

- https://github.com/edenspiekermann/a11y-dialog 
- https://github.com/gdkraus/accessible-modal-dialog 

I did not test atlaskit, but will do that and let you know what the results were.

Reply:
I captured part of this discussion in a separate issue: https://github.com/patternfly/patternfly-react/issues/1037